From abd990454c790cb9050ce41b17a84dbae4189891 Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Thu, 12 Sep 2024 16:14:32 +0800
Subject: [PATCH] support UFS storage.

Device mappings for UFS storage:
/dev/mmcblk0      => /dev/sda
/dev/mmcblk0boot0 => /dev/sdb
/dev/mmcblk0boot1 => /dev/sdc

Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 obmc-flash-bmc | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/obmc-flash-bmc b/obmc-flash-bmc
index 442bb7a..394d5ef 100644
--- a/obmc-flash-bmc
+++ b/obmc-flash-bmc
@@ -563,7 +563,7 @@ function mmc_mount() {
 
 function mmc_update() {
     # Update u-boot if needed
-    bootPartition="mmcblk0boot0"
+    bootPartition="sdb"
     devUBoot="/dev/${bootPartition}"
     imgUBoot="${imgpath}/${version}/image-u-boot"
     if [ "$(cmp_uboot "${devUBoot}" "${imgUBoot}")" != "0" ]; then
@@ -579,13 +579,13 @@ function mmc_update() {
     # by getting the partition number mmcblk0pX from their label.
     zstd -d -c "${imgpath}"/"${version}"/image-kernel | dd of="/dev/disk/by-partlabel/boot-${label}"
     number="$(readlink -f /dev/disk/by-partlabel/boot-"${label}")"
-    number="${number##*mmcblk0p}"
-    sgdisk --change-name="${number}":boot-"${label}" /dev/mmcblk0 1>/dev/null
+    number="${number##*sda}"
+    sgdisk --change-name="${number}":boot-"${label}" /dev/sda 1>/dev/null
 
     zstd -d -c "${imgpath}"/"${version}"/image-rofs | dd of="/dev/disk/by-partlabel/rofs-${label}"
     number="$(readlink -f /dev/disk/by-partlabel/rofs-"${label}")"
-    number="${number##*mmcblk0p}"
-    sgdisk --change-name="${number}":rofs-"${label}" /dev/mmcblk0 1>/dev/null
+    number="${number##*sda}"
+    sgdisk --change-name="${number}":rofs-"${label}" /dev/sda 1>/dev/null
 
     # Run this after sgdisk for labels to take effect.
     partprobe
@@ -636,11 +636,11 @@ function mmc_mirroruboot() {
     # Get current boot device; 0-primary_bootdev device; 1 - alt_bootdev
     bootdev=$(cat /sys/kernel/debug/aspeed/sbc/abr_image)
     if [[ "${bootdev}" == "0" ]]; then
-        bootPartition="mmcblk0boot0"
-        alt_bootPartition="mmcblk0boot1"
+        bootPartition="sdb"
+        alt_bootPartition="sdc"
     else
-        bootPartition="mmcblk0boot1"
-        alt_bootPartition="mmcblk0boot0"
+        bootPartition="sdc"
+        alt_bootPartition="sdb"
     fi
 
     devUBoot="/dev/${bootPartition}"
